service/src/main/java/uk/gov/mca/beacons/api/accountholder/application/MicrosoftGraphClient.java

```diff
@@ -0,0 +1,84 @@
+package uk.gov.mca.beacons.api.accountholder.application;
+
+import com.azure.identity.ClientSecretCredential;
+import com.azure.identity.ClientSecretCredentialBuilder;
+import com.microsoft.graph.authentication.TokenCredentialAuthProvider;
+import com.microsoft.graph.models.User;
+import com.microsoft.graph.requests.GraphServiceClient;
+import java.util.List;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.mca.beacons.api.accountholder.domain.AccountHolder;
+import uk.gov.mca.beacons.api.configuration.MicrosoftGraphConfiguration;
+
+@Slf4j
+@Component("microsoftGraphClient")
+public class MicrosoftGraphClient implements AuthClient {
+
+  private final ClientSecretCredential clientSecretCredential;
+  private final TokenCredentialAuthProvider tokenCredAuthProvider;
+  private final List<String> scopes = List.of(
+    "https://graph.microsoft.com/.default"
+  );
+  private final GraphServiceClient graphClient;
+
+  @Autowired
+  public MicrosoftGraphClient(MicrosoftGraphConfiguration config) {
+    this.clientSecretCredential =
+      new ClientSecretCredentialBuilder()
+        .clientId(config.getClientId())
+        .clientSecret(config.getClientSecret())
+        .tenantId(config.getB2cTenantId())
+        .build();
+    this.tokenCredAuthProvider =
+      new TokenCredentialAuthProvider(scopes, clientSecretCredential);
+    this.graphClient =
+      GraphServiceClient
+        .builder()
+        .authenticationProvider(tokenCredAuthProvider)
+        .buildClient();
+  }
+
+  public void updateUser(AccountHolder accountHolder) {
+    try {
+      User azAdUser = new User();
```

Comment:
Does account holder at this point have all the fields, or just the ones that have been updated?

If the latter, this will overwrite